extensions/vi-mode/commands.lisp

```diff
@@ -579,21 +579,25 @@
     (lem/isearch:isearch-finish)
     (lem/isearch:isearch-next)))
 
-(define-motion vi-goto-first-line () ()
+(define-motion vi-goto-first-line (&optional (n 1)) ("p")
     (:type :line
      :jump t)
-  (move-to-beginning-of-buffer)
-  (skip-whitespace-forward (current-point) t))
+  (let ((col (point-charpos (current-point))))
+    (goto-line n)
+    (move-to-column (current-point) col)))
 
 (define-motion vi-goto-line (&optional n) ("P")
     (:type :line
      :jump t)
-  (if (null n)
-      (progn
-        (move-to-end-of-buffer)
-        (line-start (current-point)))
-      (goto-line n))
-  (skip-whitespace-forward (current-point) t))
+  (let ((col (point-charpos (current-point))))
+    (if (null n)
```

Comment:
If possible, please use `cond`.

Reply:
I can't entirely agree that it improves readability in this context, but that's fine if you like it.